tapir/coop/tests/membership_resignation/test_form.py

```diff
@@ -21,30 +26,102 @@ def setUp(self) -> None:
         self.given_feature_flag_value(feature_flag_membership_resignation, True)
         mock_timezone_now(self, self.NOW)
 
-    def test_empty_form(self):
-        form = MembershipResignationForm()
-        self.assertIn("share_owner", form.fields)
-        self.assertIn("resignation_type", form.fields)
-        self.assertIn("cancellation_date", form.fields)
-        self.assertIn("transferring_shares_to", form.fields)
+    def test_membershipResignationForm_is_valid(self):
+        share_owner = ShareOwnerFactory.create()
+        resignation = MembershipResignationFactory.create()
+        data = {
+            "share_owner": share_owner,
+            "cancellation_reason": resignation.cancellation_reason,
+            "cancellation_date": resignation.cancellation_date,
+            "resignation_type": resignation.resignation_type,
+            "transferring_shares_to": resignation.transferring_shares_to,
+            "paid_out": resignation.paid_out,
+        }
+        form = MembershipResignationForm(data=data)
+        self.assertTrue(form.is_valid())
 
-    def test_validations(self):
-        self.assertRaises(
-            ValidationError, MembershipResignationForm.validate_duplicates
+    def test_validate_share_owner(self):
+        share_owner = ShareOwnerFactory.create()
+        resignation = MembershipResignationFactory.create(share_owner=share_owner)
+        form = MembershipResignationForm(data={"share_owner": resignation.share_owner})
+        form.validate_share_owner(resignation.share_owner)
+        self.assertIn("share_owner", form.errors.keys())
+        self.assertIn(
+            "This member is already resigned.",
+            form.errors["share_owner"],
+        )
+
+    def test_validate_transfer_choice(self):
+        share_owner = ShareOwnerFactory.create()
+        resignation = MembershipResignationFactory.create(
```

Comment:
Here we don't need the `MembershipResignation` to exist in the database, so we can use `build` instead of `create`. It makes the test a bit faster.